datasketches-memory-java8/src/main/java/org/apache/datasketches/memory/internal/AllocateDirect.java

```diff
@@ -111,15 +99,15 @@ StepBoolean getValid() {
     }
 
     @Override
-    public void run() throws MemoryCloseException {
+    public void run() throws IllegalStateException {
       deallocate(true);
     }
 
-    boolean deallocate(final boolean calledFromCleaner) throws MemoryCloseException {
+    boolean deallocate(final boolean calledFromCleaner) throws IllegalStateException {
       if (valid.change()) {
         if (calledFromCleaner) {
           // Warn about non-deterministic resource cleanup.
-          LOG.warning("A WritableHandle was not closed manually");
+          //LOG.warning("A direct resource was not closed explicitly");
```

Comment:
Why would this log no longer be useful to the user?

Reply:
I debated with myself how useful is this warning.  This and the one for mapping are the only uses of Logger in all of Memory.   If you feel it is still meaningful, I can put it back.

Reply:
Thinking more, I'll just put it back.  I think your questioning its removal is correct.